weldx/utility.py

```diff
@@ -562,6 +562,149 @@ def xr_interp_like(
+
+    The reference dictionary should have the dimensions as keys and those contain
+    dictionaries with the following keywords (all optional):
+
+    ``values``
+        Specify exact coordinate values to match.
+
+    ``dtype`` : str or type
+        Ensure coordinate dtype matches at least one of the given dtypes.
+
+    ``optional`` : boolean
+        default ``False`` - if ``True``, the dimension has to be in the DataArray dax
+
+    Parameters
+    ----------
+    dax:
+        xarray object which should be validated
+    ref:
+        reference dictionary
+
+    Returns
+    -------
+    bool
+        True, if the test was a success, else throws an exception
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import xarray as xr
+    >>> import weldx as wx
+    >>> dax = xr.DataArray(
+    ...     data=np.ones((3, 2, 3)),
+    ...     dims=["d1", "d2", "d3"],
+    ...     coords={
+    ...         "d1": np.array([-1, 0, 2], dtype=int),
+    ...         "d2": pd.DatetimeIndex(["2020-05-01", "2020-05-03"]),
+    ...         "d3": ["x", "y", "z"],
+    ...     }
+    ... )
+    >>> ref = dict(
+    ...     d1={"optional": True, "values": np.array([-1, 0, 2], dtype=int)},
+    ...     d2={
+    ...         "values": pd.DatetimeIndex(["2020-05-01", "2020-05-03"]),
+    ...         "dtype": ["datetime64[ns]", "timedelta64[ns]"],
+    ...     },
+    ...     d3={"values": ["x", "y", "z"], "dtype": "<U1"},
+    ... )
+    >>> wx.utility.xr_check_coords(dax, ref)
+    True
+
+    """
+    # only process the coords of the xarray
+    if isinstance(dax, (xr.DataArray, xr.Dataset)):
+        coords = dax.coords
+    else:
+        coords = dax
+
+    # check if the keys in ref are also in dax
+    _xr_valid_key(coords, ref)
+
+    for key in ref:
+        # check if the optional key is set to true
+        if "optional" in ref[key]:
+            if ref[key]["optional"] and key not in coords:
+                # skip this key - it is not in dax
+                continue
+
+        # only if the key "values" is given do the validation
+        if "values" in ref[key]:
+            if not (coords[key].values == ref[key]["values"]).all():
+                raise ValueError(f"Value mismatch in DataArray and ref['{key}']")
+
+        # only if the key "dtype" is given do the validation
+        if "dtype" in ref[key]:
+            if isinstance(ref[key]["dtype"], list):
+                if not any(
+                    _check_dtype(coords[key].dtype, var_dtype)
+                    for var_dtype in ref[key]["dtype"]
+                ):
+                    raise TypeError(
+                        f"Mismatch in the dtype of the DataArray and ref['{key}']"
+                    )
+            elif not _check_dtype(coords[key].dtype, ref[key]["dtype"]):
+                raise TypeError(
+                    f"Mismatch in the dtype of the DataArray and ref['{key}']"
+                )
```

Comment:
this looks much cleaner now @mark141  👍 
to remove final duplications:
```suggestion
        # only if the key "dtype" is given do the validation
        if "dtype" in ref[key]:
            dtype_list = ref[key]["dtype"]
            if not isinstance(dtype_list, list):
                dtype_list = [dtype_list]
            if not any(
                _check_dtype(coords[key].dtype, var_dtype) for var_dtype in dtype_list
            ):
                raise TypeError(
                    f"Mismatch in the dtype of the DataArray and ref['{key}']"
                )
```